x-pack/plugins/security_solution/public/detections/pages/detection_engine/detection_engine_no_signal_index.tsx

```diff
@@ -10,19 +10,31 @@ import { EmptyPage } from '../../../common/components/empty_page';
 import * as i18n from './translations';
 import { useKibana } from '../../../common/lib/kibana';
 
-export const DetectionEngineNoIndex = React.memo(() => {
+const DetectionEngineNoIndexComponent: React.FC<{
+  needsListsIndex: boolean;
+  needsSignalsIndex: boolean;
+}> = ({ needsListsIndex, needsSignalsIndex }) => {
   const docLinks = useKibana().services.docLinks;
+  let context: string = '';
+  if (needsSignalsIndex && needsListsIndex) {
+    context = i18n.NEEDS_SIGNALS_AND_LISTS_INDEXES;
+  } else if (needsSignalsIndex) {
+    context = i18n.NEEDS_SIGNALS_INDEX;
+  } else if (needsListsIndex) {
+    context = i18n.NEEDS_LISTS_INDEXES;
+  }
```

Comment:
You could put this into a function to remove the let and just do early returns but this is still fine as is.